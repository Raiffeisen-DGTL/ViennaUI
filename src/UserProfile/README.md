# UserProfile

Компонент для визуальной идентификации пользователя и компаний.

## Импорт

```
import { UserProfile } from 'vienna-ui';
```

## Свойства / Props

Свойства наследуются от [HTMLAttributes<HTMLAnchorElement>](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L1746)

| Prop        | Type                            | Default | Description                          |
| ----------- | ------------------------------- | ------- | ------------------------------------ |
| size        | "xs" \| "s" \| "m" \| undefined | false   |
| name        | string \| undefined             | false   | Заголовок                            |
| description | string \| undefined             | false   | Подзаголовок                         |
| photo       | string \| undefined             | false   | URL изображения                      |
| to          | string \| undefined             | false   | URL ресурса для перехода на страницу |
| align       | "left" \| "right" \| undefined  | false   | Выравнивание текста и изображения    |

## Использование

Используется в разных компонентах от шапки до карточек. Компонент состоит из родительского компонента `UserProfile` и дочернего `UserProfile.Item`. `UserProfile.Item` необходим в тех случаях, если при клике на родительский компонент нужно отобразить меню.

```
<UserProfile
    photo='https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight2&accessoriesType=Prescription01&hairColor=BlondeGolden&facialHairType=MoustacheFancy&facialHairColor=BrownDark&clotheType=BlazerSweater&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light'
    name='Super User Name'
    description='Some Description'
>
    <UserProfile.Item>Профиль</UserProfile.Item>
    <UserProfile.Item>Товары</UserProfile.Item>
    <UserProfile.Item>Предложения</UserProfile.Item>
    <UserProfile.Item>Выход</UserProfile.Item>
</UserProfile>
```

Компонент может использоваться только с аватаром, без текстового описания. Если у аватара нет или не загрузилась фотография, то вместо нее используется компонент `RoundIcon`. В таком случае символ берется с первой буквы имени пользователя или используется иконка.

```
<UserProfile name='Super User Name' description='Some Description'>
    <UserProfile.Item>Профиль</UserProfile.Item>
    <UserProfile.Item>Товары</UserProfile.Item>
    <UserProfile.Item>Предложения</UserProfile.Item>
    <UserProfile.Item>Выход</UserProfile.Item>
</UserProfile>
```

## Заголовок

##### Свойство `name`

Основной текст (имя/наименование) задается через свойство `name`.

```
<UserProfile name='Super User Name' />
```

## Подзаголовок

##### Свойство `description`

Описание/подзаголовок задается через свойство `description`.

```
<UserProfile name='Super User Name' description='Some Description' />
```

## Выравнивание

##### Свойство `align`

В некоторых случаях, может пригодится выравнивание справа через свойство `align`.

```
<UserProfile align='right' name='Super User Name' description='Some Description'>
    <UserProfile.Item>Профиль</UserProfile.Item>
    <UserProfile.Item>Товары</UserProfile.Item>
    <UserProfile.Item>Предложения</UserProfile.Item>
    <UserProfile.Item>Выход</UserProfile.Item>
</UserProfile>
```

## Профиль как ссылка

##### Свойство `to`

```
<UserProfile to='https://home.raiffeisen.ru' name='Super User Name' description='Some Description' />
```

## Размеры

##### Свойство `size`

Компонент поддерживает размеры `xs`, `s` и `m` (по-умолчанию).
