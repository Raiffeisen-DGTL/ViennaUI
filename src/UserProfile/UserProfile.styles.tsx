import styled, { css } from 'styled-components';
import { getPresets } from '../Utils/styling';
import { responsivePreset, ResponsiveProp, Breakpoints } from '../Utils/responsiveness';

const box = getPresets('userprofile', {
    base: null,
    baseHover: null,
    profile: null,
});

const profile = getPresets('userprofile.profile', {
    left: null,
    right: null,
});

const name = getPresets('userprofile.name', {
    size: responsivePreset('$size', 'm'),
});

const description = getPresets('userprofile.description', {
    base: null,
});

export interface PropsBox {
    $position: 'left' | 'right';
}
export const Box = styled.span<PropsBox>`
    position: relative;
    width: fit-content;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    box-sizing: border-box;
    cursor: pointer;
    user-select: none;
    outline: none;
    ${box.base}

    &:hover {
        ${box.baseHover}
    }

    &:focus {
        ${box.baseHover}
    }

    ${({ $position }) =>
        $position === 'right' &&
        css`
            flex-direction: row-reverse;
        `}
`;

interface PropsProfile {
    $position: 'left' | 'right';
}
export const Profile = styled.span<PropsProfile>`
    width: fit-content;

    ${({ $position }) => {
        switch ($position) {
            case 'left':
                return css`
                    ${profile.left}
                `;
            case 'right':
                return css`
                    ${profile.right}
                `;
            default:
                return null;
        }
    }}
`;

export interface Size<B = Breakpoints> {
    $size: ResponsiveProp<'xs' | 's' | 'm', B>;
}

export const Name = styled.span<Size>`
    ${name.size}
`;

export const Description = styled.span<Size>`
    display: block;
    ${description.base}

    ${({ $size }) =>
        $size !== 'm' &&
        css`
            display: none;
        `}
`;
