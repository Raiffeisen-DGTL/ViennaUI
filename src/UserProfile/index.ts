export { UserProfile, type UserProfileProps } from './UserProfile';
