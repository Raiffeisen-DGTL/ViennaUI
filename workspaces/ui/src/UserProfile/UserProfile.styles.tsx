import styled, { css } from 'styled-components';
import { getPresets } from 'vienna.ui-primitives';

const box = getPresets('userprofile', {
    base: null,
    baseHover: null,
    profile: null,
});

const profile = getPresets('userprofile.profile', {
    left: null,
    right: null,
});

const name = getPresets('userprofile.name', {
    size: 'size',
});

const description = getPresets('userprofile.description', {
    base: null,
});

export const Box = styled.span<{ position: 'left' | 'right' }>`
    position: relative;
    width: fit-content;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    box-sizing: border-box;
    cursor: pointer;
    user-select: none;
    outline: none;
    ${box.base}

    &:hover {
        ${box.baseHover}
    }

    &:focus {
        ${box.baseHover}
    }

    ${(props) =>
        props.position === 'right' &&
        css`
            flex-direction: row-reverse;
        `}
`;

export const Profile = styled.span<{ position: 'left' | 'right' }>`
    width: fit-content;

    ${(props) =>
        props.position === 'left' &&
        css`
            ${profile.left}
        `}

    ${(props) =>
        props.position === 'right' &&
        css`
            ${profile.right}
        `}
`;

export const Name = styled.span<{ size: 'xs' | 's' | 'm' }>`
    ${name.size}
`;

export const Description = styled.span<{ size: 'xs' | 's' | 'm' }>`
    display: block;
    ${description.base}

    ${(props) =>
        props.size !== 'm' &&
        css`
            display: none;
        `}
`;
