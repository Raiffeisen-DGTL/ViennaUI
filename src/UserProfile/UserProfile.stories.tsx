import React from 'react';
import { Story, Meta } from 'storybook';
import { UserProfile, UserProfileProps } from './UserProfile';

export default {
    title: 'Development/UserProfile',
    component: UserProfile,
} as Meta;

export const Overview: Story<UserProfileProps> = (args) => {
    return (
        <UserProfile
            photo='https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight2&accessoriesType=Prescription01&hairColor=BlondeGolden&facialHairType=MoustacheFancy&facialHairColor=BrownDark&clotheType=BlazerSweater&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light'
            name='Super User Name'
            description='Some Description'
            {...args}>
            <UserProfile.Item>Профиль</UserProfile.Item>
            <UserProfile.Item>Товары</UserProfile.Item>
            <UserProfile.Item>Предложения</UserProfile.Item>
            <UserProfile.Item>Выход</UserProfile.Item>
        </UserProfile>
    );
};
