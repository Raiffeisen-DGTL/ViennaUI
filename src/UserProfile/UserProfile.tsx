import React, { useState, useEffect, useCallback, ComponentProps } from 'react';
import { ManPerson } from '@fcc/icons';
import { Avatar } from '../Avatar/Avatar';
import { RoundIcon } from '../RoundIcon/RoundIcon';
import { DropList } from '../DropList/DropList';
import { Link } from './Link/Link';
import { Box, Profile, Name, Description, Size, PropsBox } from './UserProfile.styles';
import { Breakpoints } from '../Utils/responsiveness';
import { ItemProps } from '../DropList/Item/Item';
import { BoxStyled } from '../Utils/styled';

export interface UserProfileProps<B = Breakpoints>
    extends BoxStyled<typeof Box, PropsBox>,
        Pick<ComponentProps<typeof Link>, 'rel' | 'target'> {
    size?: Size<B>['$size'];
    name?: string;
    description?: string;
    photo?: string;
    to?: string;
    align?: 'left' | 'right';
}

export function UserProfile<B = void>(
    props: UserProfileProps<B extends void ? Breakpoints : B> & { Item?: React.FC<ItemProps> }
) {
    const {
        children,
        name,
        description,
        photo,
        to,
        size = 'm',
        align = 'left',
        rel,
        target,
        tabIndex = 0,
        role = 'button',
        ...attrs
    } = props;

    const [isOpen, setOpen] = useState(false);

    const close = useCallback(() => {
        setTimeout(() => setOpen(false));
        document.removeEventListener('click', close);
    }, [setOpen]);

    const open = useCallback(() => {
        setOpen((prevIsOpen) => !prevIsOpen);
        document.addEventListener('click', close);
    }, [setOpen, close]);

    const handleClickDropdown = useCallback(
        (event) => {
            event.stopPropagation();
            open();
        },
        [open]
    );

    const handleEnterKeyPress = useCallback(
        (event) => {
            event.stopPropagation();
            if (event.key === 'Enter') {
                open();
            }
        },
        [open]
    );

    useEffect(() => {
        return () => {
            document.body.removeEventListener('click', close);
        };
    }, [close]);

    const component = (
        <Box
            {...(attrs as {})}
            tabIndex={tabIndex}
            role={role}
            $position={align}
            onClick={to ? undefined : handleClickDropdown}
            onKeyPress={to ? undefined : handleEnterKeyPress}>
            <Avatar src={photo} size={size} title={name} valign='top'>
                <RoundIcon color={photo ? 'seattle10' : 'dubai10'}>{name ? name[0] : <ManPerson />}</RoundIcon>
            </Avatar>

            {name && (
                <Profile $position={align}>
                    <Name $size={size}>{name}</Name>
                    <Description $size={size}>{description}</Description>
                </Profile>
            )}

            {isOpen && <DropList float={align === 'right' ? 'end' : 'start'}>{children}</DropList>}
        </Box>
    );

    if (to) {
        return (
            <Link href={to} rel={rel} target={target}>
                {component}
            </Link>
        );
    }

    return component;
}

UserProfile.displayName = 'UserProfile';
UserProfile.Item = DropList.Item;
