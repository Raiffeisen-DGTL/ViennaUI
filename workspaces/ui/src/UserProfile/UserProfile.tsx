import React, { useState, useEffect, useCallback, HTMLAttributes } from 'react';
import { Man } from 'vienna.icons';
import Avatar from '../Avatar/Avatar';
import RoundIcon from '../RoundIcon/RoundIcon';
import DropList from '../DropList/DropList';
import { Props as ItemProps } from '../DropList/Item/Item';
import Link from './Link/Link';
import { Box, Profile, Name, Description } from './UserProfile.styles';

export interface UserProfileProps extends Omit<HTMLAttributes<HTMLAnchorElement>, 'size'> {
    [key: string]: any;
    children?: React.ReactNode;
    size?: 'xs' | 's' | 'm';
    name?: string;
    description?: string;
    photo?: string;
    to?: string;
    align?: 'left' | 'right';
}

export const UserProfile: React.FC<UserProfileProps> & { Item: React.FC<ItemProps> } = (props): JSX.Element => {
    const { children, name, description, photo, to, size = 'm', align = 'left', rel, target, ...attrs } = props;

    const [isOpen, setOpen] = useState(false);

    const close = useCallback(() => {
        setTimeout(() => setOpen(false));
        document.removeEventListener('click', close);
    }, [setOpen]);

    const open = useCallback(() => {
        setOpen((prevIsOpen) => !prevIsOpen);
        document.addEventListener('click', close);
    }, [setOpen, close]);

    const handleClickDropdown = useCallback(
        (event) => {
            event.stopPropagation();
            open();
        },
        [open]
    );

    const handleEnterKeyPress = useCallback(
        (event) => {
            event.stopPropagation();
            if (event.key === 'Enter') {
                open();
            }
        },
        [open]
    );

    useEffect(() => {
        return () => {
            document.body.removeEventListener('click', close);
        };
    }, [close]);

    const component = (
        <Box
            {...attrs}
            position={align}
            onClick={to ? undefined : handleClickDropdown}
            onKeyPress={to ? undefined : handleEnterKeyPress}>
            <Avatar src={photo} size={size} title={name} valign='top'>
                <RoundIcon color={photo ? 'seattle10' : 'dubai10'}>{name ? name[0] : <Man />}</RoundIcon>
            </Avatar>

            {name && (
                <Profile position={align}>
                    <Name size={size}>{name}</Name>
                    <Description size={size}>{description}</Description>
                </Profile>
            )}

            {isOpen && <DropList float={align === 'right' ? 'end' : 'start'}>{children}</DropList>}
        </Box>
    );

    if (to) {
        return (
            <Link href={to} rel={rel} target={target}>
                {component}
            </Link>
        );
    }

    return component;
};

UserProfile.Item = DropList.Item;
UserProfile.defaultProps = {
    size: 'm',
    align: 'left',
    tabIndex: 0,
    role: 'button',
};

UserProfile.displayName = 'UserProfile';
export default UserProfile;
