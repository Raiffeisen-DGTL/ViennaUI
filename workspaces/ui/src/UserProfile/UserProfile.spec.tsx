import React from 'react';
import UserProfile, { UserProfileProps } from './UserProfile';

test('UserProfile', () => {
    const component = (
        <UserProfile
            photo='https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight2&accessoriesType=Prescription01&hairColor=BlondeGolden&facialHairType=MoustacheFancy&facialHairColor=BrownDark&clotheType=BlazerSweater&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light'
            name='Super User Name'
            description='Some Description'>
            <UserProfile.Item>Профиль</UserProfile.Item>
            <UserProfile.Item>Товары</UserProfile.Item>
            <UserProfile.Item>Предложения</UserProfile.Item>
            <UserProfile.Item>Выход</UserProfile.Item>
        </UserProfile>
    );

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();
});

test('UserProfile w/o description', () => {
    const component = (
        <UserProfile
            photo='https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight2&accessoriesType=Prescription01&hairColor=BlondeGolden&facialHairType=MoustacheFancy&facialHairColor=BrownDark&clotheType=BlazerSweater&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light'
            name='Super User Name'>
            <UserProfile.Item>Профиль</UserProfile.Item>
            <UserProfile.Item>Товары</UserProfile.Item>
            <UserProfile.Item>Предложения</UserProfile.Item>
            <UserProfile.Item>Выход</UserProfile.Item>
        </UserProfile>
    );

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();
});

test('UserProfile w/o name and description', () => {
    const component = (
        <UserProfile photo='https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight2&accessoriesType=Prescription01&hairColor=BlondeGolden&facialHairType=MoustacheFancy&facialHairColor=BrownDark&clotheType=BlazerSweater&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light'>
            <UserProfile.Item>Профиль</UserProfile.Item>
            <UserProfile.Item>Товары</UserProfile.Item>
            <UserProfile.Item>Предложения</UserProfile.Item>
            <UserProfile.Item>Выход</UserProfile.Item>
        </UserProfile>
    );

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();
});

test('UserProfile w/o photo', () => {
    const component = (
        <UserProfile name='Super User Name' description='Some Description'>
            <UserProfile.Item>Профиль</UserProfile.Item>
            <UserProfile.Item>Товары</UserProfile.Item>
            <UserProfile.Item>Предложения</UserProfile.Item>
            <UserProfile.Item>Выход</UserProfile.Item>
        </UserProfile>
    );

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();
});

test('UserProfile w/o align', () => {
    const component = (
        <UserProfile align='right' name='Super User Name' description='Some Description'>
            <UserProfile.Item>Профиль</UserProfile.Item>
            <UserProfile.Item>Товары</UserProfile.Item>
            <UserProfile.Item>Предложения</UserProfile.Item>
            <UserProfile.Item>Выход</UserProfile.Item>
        </UserProfile>
    );

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();
});

test('UserProfile as link', () => {
    const component = (
        <UserProfile to='https://home.raiffeisen.ru' name='Super User Name' description='Some Description' />
    );

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();

    const wrapper = snapshot.mount(component);
    expect(wrapper.find('a')).toHaveLength(1);
});

test('UserProfile w/ size', () => {
    const sizes: UserProfileProps['size'][] = ['xs', 's', 'm'];

    sizes.forEach((size) => {
        const snap = snapshot.render(
            <UserProfile
                size={size}
                photo='https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight2&accessoriesType=Prescription01&hairColor=BlondeGolden&facialHairType=MoustacheFancy&facialHairColor=BrownDark&clotheType=BlazerSweater&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light'
                name='Super User Name'
                description='Some Description'>
                <UserProfile.Item>Профиль</UserProfile.Item>
                <UserProfile.Item>Товары</UserProfile.Item>
                <UserProfile.Item>Предложения</UserProfile.Item>
                <UserProfile.Item>Выход</UserProfile.Item>
            </UserProfile>
        );

        expect(snap).toMatchSnapshot(size);
    });
});

test('UserProfile w/ any props', () => {
    const props = { className: 'className', id: 'alarm', title: 'title' };
    const component = (
        <UserProfile
            {...props}
            photo='https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight2&accessoriesType=Prescription01&hairColor=BlondeGolden&facialHairType=MoustacheFancy&facialHairColor=BrownDark&clotheType=BlazerSweater&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light'
            name='Super User Name'
            description='Some Description'>
            <UserProfile.Item>Профиль</UserProfile.Item>
            <UserProfile.Item>Товары</UserProfile.Item>
            <UserProfile.Item>Предложения</UserProfile.Item>
            <UserProfile.Item>Выход</UserProfile.Item>
        </UserProfile>
    );

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();

    const wrapper = snapshot.shallow(component);
    for (const [key, value] of Object.entries(props)) {
        expect(wrapper.props()[key]).toEqual(value);
    }
});
