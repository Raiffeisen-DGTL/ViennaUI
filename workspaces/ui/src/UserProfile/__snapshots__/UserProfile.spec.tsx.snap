// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile 1`] = `
<span
  class="UserProfilestyles__Box-n702wd-0 eIMxjw"
  role="button"
  tabindex="0"
>
  <div
    class="Avatarstyles__Box-sc-2qob5o-0 ekokgP"
    title="Super User Name"
  >
    <div
      class="RoundIconstyles__Box-sc-5md3j7-0 iaWTsZ"
      color="seattle10"
    >
      S
    </div>
    <div
      class="Avatarstyles__ImageLayer-sc-2qob5o-1 cBBdOI"
      src="https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight2&accessoriesType=Prescription01&hairColor=BlondeGolden&facialHairType=MoustacheFancy&facialHairColor=BrownDark&clotheType=BlazerSweater&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light"
    />
  </div>
  <span
    class="UserProfilestyles__Profile-n702wd-1 fEQypv"
  >
    <span
      class="UserProfilestyles__Name-n702wd-2 bnHsEw"
    >
      Super User Name
    </span>
    <span
      class="UserProfilestyles__Description-n702wd-3 dSbhfc"
    >
      Some Description
    </span>
  </span>
</span>
`;

exports[`UserProfile as link 1`] = `
<a
  class="Linkstyles__Box-sc-19s6dpo-0 kqCecX"
  href="https://home.raiffeisen.ru"
>
  <span
    class="UserProfilestyles__Box-n702wd-0 eIMxjw"
    role="button"
    tabindex="0"
  >
    <div
      class="Avatarstyles__Box-sc-2qob5o-0 ekokgP"
      title="Super User Name"
    >
      <div
        class="RoundIconstyles__Box-sc-5md3j7-0 dOlcex"
        color="dubai10"
      >
        S
      </div>
      <div
        class="Avatarstyles__ImageLayer-sc-2qob5o-1 ihfKIJ"
      />
    </div>
    <span
      class="UserProfilestyles__Profile-n702wd-1 fEQypv"
    >
      <span
        class="UserProfilestyles__Name-n702wd-2 bnHsEw"
      >
        Super User Name
      </span>
      <span
        class="UserProfilestyles__Description-n702wd-3 dSbhfc"
      >
        Some Description
      </span>
    </span>
  </span>
</a>
`;

exports[`UserProfile w/ any props 1`] = `
<span
  class="UserProfilestyles__Box-n702wd-0 eIMxjw className"
  id="alarm"
  role="button"
  tabindex="0"
  title="title"
>
  <div
    class="Avatarstyles__Box-sc-2qob5o-0 ekokgP"
    title="Super User Name"
  >
    <div
      class="RoundIconstyles__Box-sc-5md3j7-0 iaWTsZ"
      color="seattle10"
    >
      S
    </div>
    <div
      class="Avatarstyles__ImageLayer-sc-2qob5o-1 cBBdOI"
      src="https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight2&accessoriesType=Prescription01&hairColor=BlondeGolden&facialHairType=MoustacheFancy&facialHairColor=BrownDark&clotheType=BlazerSweater&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light"
    />
  </div>
  <span
    class="UserProfilestyles__Profile-n702wd-1 fEQypv"
  >
    <span
      class="UserProfilestyles__Name-n702wd-2 bnHsEw"
    >
      Super User Name
    </span>
    <span
      class="UserProfilestyles__Description-n702wd-3 dSbhfc"
    >
      Some Description
    </span>
  </span>
</span>
`;

exports[`UserProfile w/ size: m 1`] = `
<span
  class="UserProfilestyles__Box-n702wd-0 eIMxjw"
  role="button"
  tabindex="0"
>
  <div
    class="Avatarstyles__Box-sc-2qob5o-0 ekokgP"
    title="Super User Name"
  >
    <div
      class="RoundIconstyles__Box-sc-5md3j7-0 iaWTsZ"
      color="seattle10"
    >
      S
    </div>
    <div
      class="Avatarstyles__ImageLayer-sc-2qob5o-1 cBBdOI"
      src="https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight2&accessoriesType=Prescription01&hairColor=BlondeGolden&facialHairType=MoustacheFancy&facialHairColor=BrownDark&clotheType=BlazerSweater&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light"
    />
  </div>
  <span
    class="UserProfilestyles__Profile-n702wd-1 fEQypv"
  >
    <span
      class="UserProfilestyles__Name-n702wd-2 bnHsEw"
    >
      Super User Name
    </span>
    <span
      class="UserProfilestyles__Description-n702wd-3 dSbhfc"
    >
      Some Description
    </span>
  </span>
</span>
`;

exports[`UserProfile w/ size: s 1`] = `
<span
  class="UserProfilestyles__Box-n702wd-0 eIMxjw"
  role="button"
  tabindex="0"
>
  <div
    class="Avatarstyles__Box-sc-2qob5o-0 XZhVf"
    title="Super User Name"
  >
    <div
      class="RoundIconstyles__Box-sc-5md3j7-0 iaWTsZ"
      color="seattle10"
    >
      S
    </div>
    <div
      class="Avatarstyles__ImageLayer-sc-2qob5o-1 cBBdOI"
      src="https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight2&accessoriesType=Prescription01&hairColor=BlondeGolden&facialHairType=MoustacheFancy&facialHairColor=BrownDark&clotheType=BlazerSweater&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light"
    />
  </div>
  <span
    class="UserProfilestyles__Profile-n702wd-1 fEQypv"
  >
    <span
      class="UserProfilestyles__Name-n702wd-2 eLyEuy"
    >
      Super User Name
    </span>
    <span
      class="UserProfilestyles__Description-n702wd-3 cWnHoF"
    >
      Some Description
    </span>
  </span>
</span>
`;

exports[`UserProfile w/ size: xs 1`] = `
<span
  class="UserProfilestyles__Box-n702wd-0 eIMxjw"
  role="button"
  tabindex="0"
>
  <div
    class="Avatarstyles__Box-sc-2qob5o-0 ihsxQj"
    title="Super User Name"
  >
    <div
      class="RoundIconstyles__Box-sc-5md3j7-0 iaWTsZ"
      color="seattle10"
    >
      S
    </div>
    <div
      class="Avatarstyles__ImageLayer-sc-2qob5o-1 cBBdOI"
      src="https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight2&accessoriesType=Prescription01&hairColor=BlondeGolden&facialHairType=MoustacheFancy&facialHairColor=BrownDark&clotheType=BlazerSweater&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light"
    />
  </div>
  <span
    class="UserProfilestyles__Profile-n702wd-1 fEQypv"
  >
    <span
      class="UserProfilestyles__Name-n702wd-2 fiUeYF"
    >
      Super User Name
    </span>
    <span
      class="UserProfilestyles__Description-n702wd-3 cWnHoF"
    >
      Some Description
    </span>
  </span>
</span>
`;

exports[`UserProfile w/o align 1`] = `
<span
  class="UserProfilestyles__Box-n702wd-0 jILASt"
  role="button"
  tabindex="0"
>
  <div
    class="Avatarstyles__Box-sc-2qob5o-0 ekokgP"
    title="Super User Name"
  >
    <div
      class="RoundIconstyles__Box-sc-5md3j7-0 dOlcex"
      color="dubai10"
    >
      S
    </div>
    <div
      class="Avatarstyles__ImageLayer-sc-2qob5o-1 ihfKIJ"
    />
  </div>
  <span
    class="UserProfilestyles__Profile-n702wd-1 iaSjVk"
  >
    <span
      class="UserProfilestyles__Name-n702wd-2 bnHsEw"
    >
      Super User Name
    </span>
    <span
      class="UserProfilestyles__Description-n702wd-3 dSbhfc"
    >
      Some Description
    </span>
  </span>
</span>
`;

exports[`UserProfile w/o description 1`] = `
<span
  class="UserProfilestyles__Box-n702wd-0 eIMxjw"
  role="button"
  tabindex="0"
>
  <div
    class="Avatarstyles__Box-sc-2qob5o-0 ekokgP"
    title="Super User Name"
  >
    <div
      class="RoundIconstyles__Box-sc-5md3j7-0 iaWTsZ"
      color="seattle10"
    >
      S
    </div>
    <div
      class="Avatarstyles__ImageLayer-sc-2qob5o-1 cBBdOI"
      src="https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight2&accessoriesType=Prescription01&hairColor=BlondeGolden&facialHairType=MoustacheFancy&facialHairColor=BrownDark&clotheType=BlazerSweater&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light"
    />
  </div>
  <span
    class="UserProfilestyles__Profile-n702wd-1 fEQypv"
  >
    <span
      class="UserProfilestyles__Name-n702wd-2 bnHsEw"
    >
      Super User Name
    </span>
    <span
      class="UserProfilestyles__Description-n702wd-3 dSbhfc"
    />
  </span>
</span>
`;

exports[`UserProfile w/o name and description 1`] = `
<span
  class="UserProfilestyles__Box-n702wd-0 eIMxjw"
  role="button"
  tabindex="0"
>
  <div
    class="Avatarstyles__Box-sc-2qob5o-0 ekokgP"
  >
    <div
      class="RoundIconstyles__Box-sc-5md3j7-0 iaWTsZ"
      color="seattle10"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          clip-rule="evenodd"
          d="M12 2a5 5 0 100 10 5 5 0 000-10zM9 7a3 3 0 116 0 3 3 0 01-6 0z"
          fill-rule="evenodd"
        />
        <path
          d="M12 13c-5.41 0-8.5 4.598-8.5 9h2c0-3.598 2.473-7 6.5-7s6.5 3.402 6.5 7h2c0-4.402-3.09-9-8.5-9z"
        />
      </svg>
    </div>
    <div
      class="Avatarstyles__ImageLayer-sc-2qob5o-1 cBBdOI"
      src="https://avataaars.io/?avatarStyle=Transparent&topType=LongHairStraight2&accessoriesType=Prescription01&hairColor=BlondeGolden&facialHairType=MoustacheFancy&facialHairColor=BrownDark&clotheType=BlazerSweater&eyeType=Default&eyebrowType=Default&mouthType=Default&skinColor=Light"
    />
  </div>
</span>
`;

exports[`UserProfile w/o photo 1`] = `
<span
  class="UserProfilestyles__Box-n702wd-0 eIMxjw"
  role="button"
  tabindex="0"
>
  <div
    class="Avatarstyles__Box-sc-2qob5o-0 ekokgP"
    title="Super User Name"
  >
    <div
      class="RoundIconstyles__Box-sc-5md3j7-0 dOlcex"
      color="dubai10"
    >
      S
    </div>
    <div
      class="Avatarstyles__ImageLayer-sc-2qob5o-1 ihfKIJ"
    />
  </div>
  <span
    class="UserProfilestyles__Profile-n702wd-1 fEQypv"
  >
    <span
      class="UserProfilestyles__Name-n702wd-2 bnHsEw"
    >
      Super User Name
    </span>
    <span
      class="UserProfilestyles__Description-n702wd-3 dSbhfc"
    >
      Some Description
    </span>
  </span>
</span>
`;
